File: app/src/main/java/com/themaker/fshmo/klassikaplus/data/domain/Item.java
package com.themaker.fshmo.klassikaplus.data.domain;

public class Item {

    private String id;
    private String extId;// TODO: 2/6/2019 add extId to DB
    private String name; //h1
    private String description; //ключевое описание
    private String vendorCode;
    private Boolean novelty;
    private String descriptionLong;
    private String pageAlias;
    private String icon;
    private String photo;
    private String manufacturer; //производитель
    private Double basePrice;
    private Double discount;
    private Boolean discountable;
    private Double price;

    @Override
    public String toString() {
        return "Item{" +
                "id='" + id + '\'' +
                ", extId='" + extId + '\'' +
                ", name='" + name + '\'' +
                ", description='" + description + '\'' +
                ", icon='" + icon + '\'' +
                ", price='" + price + '\'' +
                '}';
    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getVendorCode() {
        return vendorCode;
    }

    public void setVendorCode(String vendorCode) {
        this.vendorCode = vendorCode;
    }

    public Boolean getNovelty() {
        return novelty;
    }

    public void setNovelty(Boolean novelty) {
        this.novelty = novelty;
    }

    public String getDescriptionLong() {
        return descriptionLong;
    }

    public void setDescriptionLong(String descriptionLong) {
        this.descriptionLong = descriptionLong;
    }

    public String getPageAlias() {
        return pageAlias;
    }

    public void setPageAlias(String pageAlias) {
        this.pageAlias = pageAlias;
    }

    public String getIcon() {
        return icon;
    }

    public void setIcon(String icon) {
        this.icon = icon;
    }

    public String getPhoto() {
        return photo;
    }

    public void setPhoto(String photo) {
        this.photo = photo;
    }

    public String getManufacturer() {
        return manufacturer;
    }

    public void setManufacturer(String manufacturer) {
        this.manufacturer = manufacturer;
    }

    public Double getBasePrice() {
        return basePrice;
    }

    public void setBasePrice(double basePrice) {
        this.basePrice = basePrice;
    }

    public Double getDiscount() {
        return discount;
    }

    public void setDiscount(double discount) {
        this.discount = discount;
    }

    public Double getPrice() {
        return price;
    }

    public void setPrice(double price) {
        this.price = price;
    }

    public Boolean getDiscountable() {
        return discountable;
    }

    public void setDiscountable(Boolean discountable) {
        this.discountable = discountable;
    }
}

File: app/src/main/java/com/themaker/fshmo/klassikaplus/data/domain/Item.kt
package com.themaker.fshmo.klassikaplus.data.domain

import android.os.Parcel
import android.os.Parcelable
import java.io.Serializable

data class Item(

    var id: String,
    var extId: String,
    var name: String,
    var description: String,
    var vendorCode: String,
    var novelty: Boolean,
    var descriptionLong: String,
    var pageAlias: String,
    var icon: String,
    var photo: String,
    var manufacturer: String,
    var basePrice: Double,
    var discount: Double,
    var discountable: Boolean,
    var price: Double,
    var category: String,
    var categoryId: Int

) : Serializable, Parcelable {
    override fun toString(): String {
        return "Item{" +
                "id='" + id + '\''.toString() +
                ", extId='" + extId + '\''.toString() +
                ", name='" + name + '\''.toString() +
                ", description='" + description + '\''.toString() +
                ", icon='" + icon + '\''.toString() +
                ", price='" + price + '\''.toString() +
                '}'.toString()
    }

    constructor(source: Parcel) : this(
        source.readString(),
        source.readString(),
        source.readString(),
        source.readString(),
        source.readString(),
        1 == source.readInt(),
        source.readString(),
        source.readString(),
        source.readString(),
        source.readString(),
        source.readString(),
        source.readDouble(),
        source.readDouble(),
        1 == source.readInt(),
        source.readDouble(),
        source.readString(),
        source.readInt()
    )

    override fun describeContents() = 0

    override fun writeToParcel(dest: Parcel, flags: Int) = with(dest) {
        writeString(id)
        writeString(extId)
        writeString(name)
        writeString(description)
        writeString(vendorCode)
        writeInt((if (novelty) 1 else 0))
        writeString(descriptionLong)
        writeString(pageAlias)
        writeString(icon)
        writeString(photo)
        writeString(manufacturer)
        writeDouble(basePrice)
        writeDouble(discount)
        writeInt((if (discountable) 1 else 0))
        writeDouble(price)
        writeString(category)
        writeInt(categoryId)
    }

    companion object {
        @JvmField
        val CREATOR: Parcelable.Creator<Item> = object : Parcelable.Creator<Item> {
            override fun createFromParcel(source: Parcel): Item = Item(source)
            override fun newArray(size: Int): Array<Item?> = arrayOfNulls(size)
        }
    }
}

